src/pepr/operator/controllers/network/policies.ts

```diff
@@ -0,0 +1,103 @@
+import { K8s, Log, kind } from "pepr";
```

Comment:
NetworkPolcies are created after a UDSPackage is created, which is in parallel to the rest of the app's installation into it's namespace. Nothing blocks the execution of the app's Pods, they may run before the NetworkPolcies are finished being created.

Not sure if/how we want to handle this, but it's a general risk of doing policy enforcement via an operator CR that's installed parallel to an app. You _could_ have the admission controller deny Pods until the UDSPackage is status ready. But you'd also have to do things like make sure Zarf installs the UDSPackage before other resources in the namespace.

(or maybe I've missed something. I'm not fully tracking how the UDSPackage CR gets installed, is it via Helm just as another resource beside the app?)

Reply:
It's not exactly parallel, but it's also not entirely guaranteed that everything is processed before the app is deployed. Using [grafana](https://github.com/defenseunicorns/uds-core/blob/uds-operator/src/grafana/zarf.yaml#L12-L15) as an example, the grafana-config chart contains the UDSPackage CR and other configuration (in this case a configmap for grafana datasources), but will not have any pods. After install is complete then the upstream grafana helm chart is installed.

The missing bit is waiting on the UDSPackage CR to be "rolled out" (i.e. all network policies, etc applied). That could be done with a zarf wait action on `status.phase` but that would require the two charts to be split into separate components, which would make the imports in the [uds-core package](https://github.com/defenseunicorns/uds-core/blob/uds-operator/packages/standard/zarf.yaml) double. I think in the broadest sense "it works" as is but that's not guaranteed strictly.

Reply:
We discussed this and have decided that moving the policies to before the istio injection ensures the fastest policy generation and the risk is narrow enough to not address further at this time (would only affect egress).